### PR TITLE
Fixed the sign in and sign up feature

### PR DESCRIPTION
**Pull request by JulesHerrscher**
Fixed the login feature that required the user to click two times instead of 1 and also fixed a crash when either the password or the email was empty, as well as the sign up that made the application crash when the email was empty. The problem was the time it took to fetch the user from the database, I added a callback in order to solve this problem. I also managed to print the different error messages that I get through the callback. It would help the user to know where he is failing when he wants to connect.

### CODE REVIEW

File: app/src/androidTest/java/com/swent/assos/SigninTest.kt [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+            performTextInput("jules.herrscher@icloud.com")
+          }
+          passwordField {
+            assertIsDisplayed()
+            performTextInput("password")
+          }
+        }
+      }
+    }
+  }
```
Comment (Tikipiou) [inline app/src/androidTest/java/com/swent/assos/SigninTest.kt:128]: Check for an error message.

File: app/src/main/java/com/swent/assos/ui/login/LoginScreen.kt [hunk 2]
```diff
@@ -45,6 +46,7 @@ fun LoginScreen(navigationActions: NavigationActions) {
         var password by remember { mutableStateOf("") }
         val loginViewModel: LoginViewModel = hiltViewModel()
         var userNotFound by remember { mutableStateOf(false) }
+        var errorMessage by remember { mutableStateOf("") }
```
Comment (Tikipiou) [inline app/src/main/java/com/swent/assos/ui/login/LoginScreen.kt:49]: Create the states `userNotFound` and  `errorMessage` in the `viewModel` and set its value directly in the method `signIn`

File: app/src/main/java/com/swent/assos/ui/login/LoginScreen.kt [hunk 3]
```diff
@@ -53,32 +55,46 @@ fun LoginScreen(navigationActions: NavigationActions) {
...
+              } else {
+                loginViewModel.signIn(email, password) { e ->
+                  Log.d("LoginScreen", "User: ${loginViewModel.user}")
+                  if (loginViewModel.user != User("", "", "", "", emptyList(), emptyList())) {
+                    navigationActions.navigateTo(Destinations.HOME.route)
+                  } else {
+                    userNotFound = true
+                    errorMessage = e?.message ?: "User not found"
+                  }
+                }
```
Comment (Tikipiou) [inline app/src/main/java/com/swent/assos/ui/login/LoginScreen.kt:90]: Do all of this directly in the `signIn`method.

File: app/src/main/java/com/swent/assos/ui/login/SignUpScreen.kt [hunk 4]
```diff
@@ -41,7 +41,7 @@ fun SignUpScreen(navigationActions: NavigationActions) {
   var confirmPassword by remember { mutableStateOf("") }
   val loginViewModel: LoginViewModel = hiltViewModel()
   var error by remember { mutableStateOf("") }
-  var passwordTooShort by remember { mutableStateOf(false) }
+  var badCredentials by remember { mutableStateOf(false) }
```
Comment (Tikipiou) [inline app/src/main/java/com/swent/assos/ui/login/SignUpScreen.kt:44]: Same as before, create it in the `viewModel`and set its value in the `signUp`method.

**Review by Tikipiou — CHANGES_REQUESTED** [review]
Globally good, except for some state in the UI layer instead of the ViewModel layer.

File: app/src/androidTest/java/com/swent/assos/SigninTest.kt (hunk 1 above)
Comment (JulesHerrscher) [reply app/src/androidTest/java/com/swent/assos/SigninTest.kt:128]: Thank you ! Fixed it.

File: app/src/main/java/com/swent/assos/ui/login/LoginScreen.kt (hunk 2 above)
Comment (JulesHerrscher) [reply app/src/main/java/com/swent/assos/ui/login/LoginScreen.kt:49]: Thank you ! I fixed it

**Review by Tikipiou — APPROVED** [review]

**Pull request closed without merging**